### PR TITLE
add missing GH token with correct permissions

### PR DESCRIPTION
**Pull request by ab77**
* fixes [this](https://github.com/balena-io/environment-production/actions/runs/8990387898/job/24695669680#step:8:51)

change-type: patch

### CODE REVIEW

**Review by klutchell — APPROVED** [review]

**Comment by ab77** [comment]
.. actually, I am a bit lost as to why [this](https://github.com/product-os/flowzone/actions/runs/8986693028/job/24683489154#step:8:53) works, but [this](https://github.com/balena-io/environment-production/actions/runs/8990387898/job/24695669680#step:8:50) doesn't..

as fas as I can tell, they are both running the same code..

File: flowzone.yml [hunk 1]
```diff
@@ -1367,6 +1367,7 @@ jobs:
           github.event.pull_request.merged == true
           && steps.create_tag.outputs.sha != ''
         env:
+          GH_TOKEN: ${{ steps.gh_app_token.outputs.token || secrets.FLOWZONE_TOKEN }}
```
Comment (klutchell) [inline flowzone.yml:1370]: [This](https://github.com/balena-io/environment-production/blob/fb45587920dd21973295af67b1ba523f525fd1d3/.github/workflows/flowzone.yml#L11-L24) is why

**Review by klutchell — CHANGES_REQUESTED** [review]

File: flowzone.yml (hunk 1 above)
Comment (ab77) [reply flowzone.yml:1370]: yep, I get it now, the default token permissions are used otherwise, which in that workflow are all `read`

Comment (klutchell) [reply flowzone.yml:1370]: So we can close this PR right, and just adjust the permissions of the automatic GITHUB_TOKEN on that repo?

I would prefer we use the automatic token where it makes sense to do so, and only use the Flowzone token for cases where special permissions are required (like cross-repo permissions)

**Comment by ab77** [comment]
I'll adjust the token in the repo.

**Pull request closed without merging**